modules/data-entry/src/main/java/io/uhndata/cards/dataentry/internal/serialize/QuestionnaireToCsvProcessor.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.uhndata.cards.dataentry.internal.serialize;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.json.JsonArray;
+import javax.json.JsonException;
+import javax.json.JsonObject;
+import javax.json.JsonString;
+import javax.json.JsonValue;
+import javax.json.JsonValue.ValueType;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.sling.api.resource.Resource;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.uhndata.cards.dataentry.internal.serialize.labels.ResourceCSVProcessor;
+
+/**
+ * CSV serializer that can process Questionnaires.
+ *
+ * @version $Id$
+ */
+@Component(service = ResourceCSVProcessor.class)
+public class QuestionnaireToCsvProcessor implements ResourceCSVProcessor
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(QuestionnaireToCsvProcessor.class);
+
+    @Override
+    public boolean canProcess(final Resource resource)
+    {
+        return resource.isResourceType("cards/Questionnaire");
+    }
+
+    @Override
+    public String serialize(Resource resource)
+    {
+        // The proper serialization depends on "deep", "dereference" and "labels", but we may allow other JSON
+        // processors to be enabled/disabled to further customize the data, so we also append the original selectors
+        final String processedPath = resource.getPath()
+            + resource.getResourceMetadata().getResolutionPathInfo() + ".deep.dereference.labels";
```

Comment:
```suggestion
            + resource.getResourceMetadata().getResolutionPathInfo() + ".deep.dereference.-labels";
```
Deactivate labels by default.

Reply:
To be postponed, not necessary in the first iterration.